wger.py

```diff
@@ -127,186 +161,188 @@ def create_exercise(self):
+    
+    def delete_day(self,workout,day):
+        """This function delete a training day"""
+        id_workout=self.match(self.get_info('workout/').get('results'),workout,'name','id')
+        days=[i for i in self.get_info('day/').get('results') if i['training']==id_workout]
+        id_day=self.match(days,day,'description','id')
+        if id_day=="":
+            return (False,"Day not found")
+        self.delete_info(f"day/{id_day}")
+        return any([i['id'] for i in self.get_info('day/').get('results') if i['id']==id_day])==False
 
     def delete_workout(self,workout_name):
         """This function delete an workout"""
-        workouts=self.get_info('workout/')['results']
-        id=""
-        for i in workouts:
-            if i['name']==workout_name:
-                id=i['id']
+        workouts=self.get_info('workout/').get('results')
+        id=self.match(workouts,workout_name,'name','id')
         if id=="":
-            return "Workout not found"
-        self.delete_info(f'workout/{id}/')
-        return f"Workout {workout_name} was deleted successfully!"
+            return (False,"Workout not found")
+        self.delete_info(f'workout/{id}')
+        return any([i['name'] for i in self.get_info('workout/').get('results') if i['name']==workout_name])==False
 
     def save_ex_details(self,dir_name):
-        ex_images = []
-        ex_comments = []
-        all_exercises=[]
-        exercises = [i['exercise'] for i in user1.get_info('setting')['results']]
-        for i in range(20, 100, 20):
-            ex_images.extend(user1.get_info(f'exerciseimage/?limit=20&offset={i}')['results'])
-        for i in range(20, 120, 20):
-            ex_comments.extend(user1.get_info(f'exercisecomment/?limit=20&offset={i}')['results'])
-        for i in range(20, 400, 20):
-            all_exercises.extend(user1.get_info(f'exercise/?limit=20&offset={i}')['results'])
+        exercises = [i['exercise'] for i in user1.get_info('setting/').get('results')]
+        ex_images=self.get_info(f'exerciseimage/',100).get('results')
+        ex_comments=self.get_info(f'exercisecomment/',120).get('results')
+        all_exercises=self.get_info(f'exercise/',400).get('results')
 
         for i in exercises:
             ex_image = "This exercise has no image available."
             ex_comment = "This exercise has no comments available."
             ex_name="Exercise"
             for j in all_exercises:
                 if j['id']==i:
-                    ex_name=j['name']
+                    ex_name=j.get('name')
             for j in ex_images:
                 if j['id'] == i:
-                    ex_image = j['image']
+                    ex_image = j.get('image')
             for j in ex_comments:
                 if j['id'] == i:
-                    ex_comment = j['comment']
+                    ex_comment = j.get('comment')
             f = open(f'{dir_name}\exercise_{i}.html', 'w')
```

Comment:
Read about context managers. They are the preferred way of working with open() in order to avoid ungraceful program breaks that would leave that file malformed.